spring-integration-redis/src/main/java/org/springframework/integration/redis/util/RedisLockRegistry.java

```diff
@@ -364,9 +365,9 @@ private boolean subscribeLock(long time) throws ExecutionException, InterruptedE
 				return true;
 			}
 
-			if (!RedisLockRegistry.this.redisMessageListenerContainer.isRunning()) {
-				RedisLockRegistry.this.redisMessageListenerContainer.afterPropertiesSet();
-				RedisLockRegistry.this.redisMessageListenerContainer.start();
+			if (!(RedisLockRegistry.this.isRunningRedisMessageListenerContainer
+					&& RedisLockRegistry.this.redisMessageListenerContainer.isRunning())) {
```

Comment:
I think just one our `isRunningRedisMessageListenerContainer` is enough.
The problem is really that `RedisMessageListenerContainer` sets `running = true;` in the beginning of `start()`.
Better to have it in the end, when subscription is ready.
But still we need that `synchronized (RedisLockRegistry.this.redisMessageListenerContainer)` anyway to block any concurrent calls for the `start()`.

WDYT?